.github/workflows/main.yml

```diff
@@ -0,0 +1,35 @@
+name: Check for dependency upgrades
+
+on:
+  schedule:
+    - cron: '30 5 * * 0'
+  
+jobs:
+  upgrade-dependencies:
+    name: Check for dependency upgrades
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Activate a virtual environment
+        run: python -m venv .venv && source .venv/bin/activate
+      - name: Install pip-upgrader
+        run: pip install pip-upgrader
+      - name: Update python dependencies
+        run: |
+          find ./docker -name requirements.txt -exec pip-upgrade {} --skip-package-installation -p all \;
+      - name: Update node dependencies
+        run: |
+          source .venv/bin/activate && \
+          pip install nodeenv && \
+          nodeenv -p && \
+          npm install -g npm-check-updates && \
+          find ./docker -name package.json -exec ncu --packageFile {} -u \;
+      # TODO: push to the proper branch after pr approved
+      - name: Push to main branch
+        run: |
+          git config --global user.name "xmedr"
+          git config --global user.email "xavier.medrano@datamade.us"
+
+          git add .
+          git commit -m "Update pypi and node dependencies"
+          git push origin feature/dependency-auto-update
```

Comment:
We should use `${{ github.ref_name }}` here, or some other value from the action context: https://docs.github.com/en/actions/learn-github-actions/contexts

That way you don't have to change it. This assumes that once this code is on the main branch and the cron job runs, it'll use the value in `${{ github.ref_name }}`, which is what we want, right?

Reply:
Ah cool I wasn't aware of those context variables, that's exactly what we'd want ty! I've got that change in

Reply:
nice!